ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateSignatureProvider.fs

```diff
@@ -0,0 +1,312 @@
+
+                    sb.Append(getName memberDecl) |> ignore
+                    sb.Append(": ") |> ignore
+
+                    let symbolUse = memberDecl.GetFcsSymbolUse()
+                    if isNotNull symbolUse then
+                        let mfv = symbolUse.Symbol.As<FSharpMemberOrFunctionOrValue>()
+                        if isNotNull mfv then
+                            if memberDecl.IsStatic then
+                                sb.Append(mfv.FullType.Format(symbolUse.DisplayContext)) |> ignore
+                            else
+                                // mfv.FullType will contain the type of the instance, so we cannot use that.
+                                let parameters =
+                                    mfv.CurriedParameterGroups
+                                    |> Seq.map (fun parameterGroup ->
+                                        parameterGroup
+                                        |> Seq.map (fun parameter -> parameter.Type.Format(symbolUse.DisplayContext))
+                                        |> String.concat " * "
+                                    )
+                                    |> String.concat " -> "
+                                let returnType = mfv.ReturnParameter.Type.Format(symbolUse.DisplayContext)
+                                sb.Append($"{parameters} -> {returnType}") |> ignore
+
+                    sb.ToString()
+
+                factory.CreateTypeMemberSignature(sourceString)
+            | _ -> null
+
+        for decl in fsharpFile.ModuleDeclarations do
+            let signatureModule : IModuleLikeDeclaration =
+                match decl with
+                | :? INamedModuleDeclaration as nmd ->
+                    factory.CreateModule(nmd.DeclaredElement.GetClrName().FullName)
+                | :? IGlobalNamespaceDeclaration ->
+                    factory.CreateNamespace("global") :?> _
+                | :? INamedNamespaceDeclaration as nnd ->
+                    // TODO: add an interface that could unify named and global namespace.
+                    factory.CreateNamespace(nnd.QualifiedName) :?> _
+                | decl -> failwithf $"Unexpected declaration, got: %A{decl}"
+
+            ModificationUtil.AddChildAfter(signatureModule.LastChild, NewLine(lineEnding)) |> ignore
+            let signatureModule = processModuleLikeDeclaration 0 decl signatureModule
+            ModificationUtil.AddChild(signatureFile, signatureModule) |> ignore
+
+        signatureFile
+
+    override this.IsAvailable(context: FSharpGeneratorContext): bool =
+        let node = context.Root :?> IFSharpTreeNode
+        let currentFSharpFile = node.FSharpFile
+        if currentFSharpFile.IsFSharpSigFile() then false else
+
+        let solution = node.GetSolution()
+        let isSettingEnabled = solution.IsFSharpExperimentalFeatureEnabled(ExperimentalFeature.GenerateSignatureFile)
+        if not isSettingEnabled then false else
```

Comment:
It seems this place is being checked a bit too late: the workflow gets shown inside Alt+Enter menu and is just getting disabled. We could try to use the `dataContext` that is passed to `FSharpGenerateSignatureWorkflowProvider.CreateWorkflow`, get a solution from it and check if the setting is enabled.

Reply:
Good catch, I think I solved it.